context.go

```diff
@@ -319,3 +290,21 @@ func ValidateContextName(context string) error {
 	}
 	return nil
 }
+
+func (ctxMap *ContextMap) Set(key, value string) error {
+	contextFilePath := createFilePath(viper.GetString(config.CurrentContext) + ".yaml")
+	file, err := utils.DecodeYAMLFile(contextFilePath)
+	if err != nil {
+		return err
+	}
+
+	if key == config.EnvFnAPIURL {
+		err := ValidateAPIURL(value)
+		if err != nil {
+			return err
+		}
+	}
+
+	file[key] = value
+	return config.WriteYamlFile(contextFilePath, file)
```

Comment:
<- here you would open a file one more time to write back a new content, so what i recommend it to change the signature of these two calls to accept `os.File` to do less redundant IO operations here. But it's up to you.